### PR TITLE
[Feature][connector elasticsearch] sink support write object 

### PR DESCRIPTION
**Pull request by cccc0212**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by EricJoy2048** [comment]
Please add e2e to test this pr.

**Comment by cccc0212** [comment]
@EricJoy2048 I have added e2e to test this pr.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/FakeToElasticsearchIT.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    @TestTemplate
+    public void testFakeToElasticsearch(TestContainer container)
+            throws IOException, InterruptedException {
+        Container.ExecResult execResult =
+                container.executeJob("/elasticsearch/fake_to_elasticsearch.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+        List<String> sinkData = readSinkData();
+        // old es value: c_json_string : "{\"key\":\"value\"}"
+        // new es value: c_json_string : {"key":"value"}
+        System.out.println(sinkData.toString());
```
Comment (EricJoy2048) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-elasticsearch-e2e/src/test/java/org/apache/seatunnel/e2e/connector/elasticsearch/FakeToElasticsearchIT.java]: Please use `Assertions` to assert the result.

File: seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java [hunk 2]
```diff
@@ -171,7 +171,20 @@ private Map<String, Object> toDocumentMap(SeaTunnelRow row) {
...
+                            doc.put(fieldNames[i], o);
+                        } else {
+                            doc.put(fieldNames[i], value);
+                        }
+                    } else {
+                        doc.put(fieldNames[i], value);
+                    }
+                } else {
+                    doc.put(fieldNames[i], value);
+                }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/serialize/ElasticsearchRowSerializer.java:187]:
```suggestion
                if (value != null && value instanceof String) {
                    Object o = convertToObjectOrArray(value.toString());
                    if (o != null) {
                        doc.put(fieldNames[i], o);
                    } else {
                        doc.put(fieldNames[i], value);
                    }
                } else {
                    doc.put(fieldNames[i], value);
                }
```

**Comment by Hisoka-X** [comment]
I am skeptical about the function this PR wants to achieve. If the upstream declares a String, we cannot force it to be converted to an object type. This is illogical. If he wants the data to be converted to object type, he should define the data as object type at the original end. Or use transform to convert data type.

**Comment by cccc0212** [comment]
How to convert data to object type if the source is MySQL

**Comment by Hisoka-X** [comment]
> How to convert data to object type if the source is MySQL

1. we should implement new function to convert string to row in SQL function. 2. user can implement their own UDF in SQL transform

**Comment by hailin0** [comment]
> > How to convert data to object type if the source is MySQL
> 
> 1. we should implement new function to convert string to row in SQL function. 2. user can implement their own UDF in SQL transform

agree

**Comment by hailin0** [comment]
No update, closed this pr

**Pull request closed without merging**